src/tools/blur/blurtool.cpp

```diff
@@ -16,73 +16,127 @@
 //     along with Flameshot.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "blurtool.h"
-#include <QPainter>
+#include <QApplication>
 #include <QGraphicsBlurEffect>
 #include <QGraphicsPixmapItem>
 #include <QGraphicsScene>
-#include <QApplication>
+#include <QImage>
+#include <QPainter>
 
-BlurTool::BlurTool(QObject *parent) : AbstractTwoPointTool(parent) {
+BlurTool::BlurTool(QObject *parent) : AbstractTwoPointTool(parent) {}
 
+QIcon BlurTool::icon(const QColor &background, bool inEditor) const
+{
+  Q_UNUSED(inEditor);
+  return QIcon(iconPath(background) + "blur.svg");
 }
+QString BlurTool::name() const { return tr("Blur"); }
 
-QIcon BlurTool::icon(const QColor &background, bool inEditor) const {
-    Q_UNUSED(inEditor);
-    return QIcon(iconPath(background) + "blur.svg");
-}
-QString BlurTool::name() const {
-    return tr("Blur");
-}
+QString BlurTool::nameID() { return QLatin1String(""); }
 
-QString BlurTool::nameID() {
-    return QLatin1String("");
+QString BlurTool::description() const
+{
+  return tr("Set Blur as the paint tool");
 }
 
-QString BlurTool::description() const {
-    return tr("Set Blur as the paint tool");
-}
+CaptureTool *BlurTool::copy(QObject *parent) { return new BlurTool(parent); }
+
+void write_block(QImage &image, int x_start, int y_start, int pixel_size,
+                 QRgb block_color)
+{
+  assert(x_start + pixel_size < image.width());
+  assert(y_start + pixel_size < image.height());
 
-CaptureTool* BlurTool::copy(QObject *parent) {
-    return new BlurTool(parent);
+  for (auto x = x_start; x < x_start + pixel_size; x++)
+  {
+    for (auto y = y_start; y < y_start + pixel_size; y++)
+    {
+      image.setPixel(x, y, block_color);
+    }
+  }
 }
 
-void BlurTool::process(QPainter &painter, const QPixmap &pixmap, bool recordUndo) {
-    if (recordUndo) {
-        updateBackup(pixmap);
+QRgb calculate_block_averge(QImage &image, int x_start, int y_start,
+                            int pixel_size)
+{
+  assert(x_start + pixel_size < image.width());
+  assert(y_start + pixel_size < image.height());
+
+  auto red_count = 0;
```

Comment:
I'm not very convinced on the use of auto when the type is trivial. I would prefer more explicit code.

Reply:
Fixed in latest push.